src/osd/OSD.cc

```diff
@@ -3809,8 +3834,6 @@ void OSD::add_newly_split_pg(PG *pg, PG::RecoveryCtx *rctx)
     }
     peering_wait_for_split.erase(to_wake);
   }
-  if (!service.get_osdmap()->have_pg_pool(pg->pg_id.pool()))
-    _remove_pg(pg);
```

Comment:
No replacement?

Reply:
i need to review all of the split vs pg_map code; haven't done that yet.

Reply:
added todo

Reply:
ok, it turns out the current split tracking code is fine.  This can go away along with some cleanup helpers because post-split pg deletion is now handled when the new/child pg consumes the osdmap with the pool deletion (i.e., pg removal is less of a special case).

Reply:
Ah yeah, I figured that out in one of the other code paths...much nicer!